lightning/src/ln/channel.rs

```diff
@@ -3127,6 +3130,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		self.our_htlc_minimum_msat
 	}
 
+	/// Allowed in any state (including after shutdown)
+	pub fn get_our_htlc_maximum_msat(&self) -> u64 {
+		let mut result = (self.channel_value_satoshis as f64 * 0.9).round() as u64; // upper bound by channel capacity
+		if let Some(configured_our_htlc_maximum_msat) = self.our_htlc_maximum_msat {
+			result = cmp::min(configured_our_htlc_maximum_msat, result)
```

Comment:
I think we should also take the min with max_htlc_value_in_flight_msat, since obviousl no single HTLC can violate the total in-flight limit.